app/models/conversation.rb

```diff
@@ -0,0 +1,43 @@
+# == Schema Information
+#
+# Table name: conversations
+#
+#  id         :bigint           not null, primary key
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+class Conversation < ApplicationRecord
+  has_many :conversation_users, dependent: :destroy
+  has_many :users, through: :conversation_users
+  has_many :messages, dependent: :destroy
+
+  def self.create_conversations(matching_users)
+    user = matching_users.pop
+    matching_users.each do |match|
+      conversation = share_conversation(user, match)
+      if conversation.blank?
+        conversation = Conversation.create!
+        user.conversations << conversation
+        match.conversations << conversation
+      end
+      conversation.messages.create!(
+        text: I18n.t('api.messages.target_match')
+      )
+    end
+  end
+
+  def self.share_conversation(user, match)
```

Comment:
Can't you use queries for this? A scope for example? `scope :between, -> (user_a, user_b) { where.... }`
and then you can call `Coversation.between(user, match)`
Also, using a query will have better performance than making this on memory.

Reply:
Will use detect instead, as we spoke earlier.